lib/sql_footprint/sql_anonymizer.rb

```diff
@@ -1,17 +1,23 @@
 module SqlFootprint
   class SqlAnonymizer
-    GSUBS = {
+    ANON_REGEXES = {
       /\sIN\s\(.*\)/ => ' IN (values-redacted)'.freeze, # IN clauses
       /([\s\(])'.*'/ => "\\1'value-redacted'".freeze, # literal strings
       /N''.*''/ => "N''value-redacted''".freeze, # literal MSSQL strings
-      /\s+(!=|=|<|>|<=|>=)\s+[0-9]+/ => ' \1 number-redacted', # numbers
-      /\s+VALUES\s+\(.+\)/ => ' VALUES (values-redacted)', # VALUES
-    }.freeze
+      /\s+(!=|=|<|>|<=|>=)\s+[0-9]+/ => ' \1 number-redacted'.freeze, # numbers
+      /\s+VALUES\s+\(.+\)/ => ' VALUES (values-redacted)'.freeze, # VALUES
+    }
 
     def anonymize sql
-      GSUBS.reduce(sql) do |s, (regex, replacement)|
+      ANON_REGEXES.reduce(sql) do |s, (regex, replacement)|
         s.gsub regex, replacement
       end
     end
+
+    class << self
+      def add_rule regex, replacement
+        ANON_REGEXES.merge!(regex => replacement)
```

Comment:
This is equivalent and slightly more straightforward:

```ruby
ANON_REGEXES[regex] = replacement
```

Reply:
So, there seems to be mixed opinions on this approach... What should I do?

Reply:
@drm2 @mikegee It’s [really] bad practice to mutate a constant.

Reply:
Mutating global state at runtime is a bad practice. All ruby programs mutate global state during bootup (defining classes, for instance). Configuring libraries also mutates global state.

We can freeze this hash when the client app calls `SqlFootprint.start`. Is that sufficient or should we pursue refactoring such that this global state mutation isn't backed by a constant to appease casual observers?

Reply:
@mikegee @schneiderderek I could also just use a class variable instead of a constant. Thoughts?